romancal/ramp_fitting/ramp_fit_step.py

```diff
@@ -45,67 +39,36 @@ class RampFitStep (Step):
     reference_file_types = ['readnoise', 'gain']
 
     def process(self, input):
-
-        # Open input as a Roman DataModel (single integration; 3D arrays)
-        with Roman.datamodels.RampModel(input) as R_input_model:  # syntax ?
-
-            # Create a JWST DataModel (4D arrays) to populate
-            input_model = JWST.datamodels.RampModel()  # syntax ?
-
-            # Populate JWST model with correct sized arrays from Roman model
-            input_model.meta = R_input_model.meta
-            input_model.data = np.broadcast_to(R_input_model.data, (1,) +
-                                               R_input_model.data.shape)
-
-            input_model.pixeldq = R_input_model.pixeldq
-
-            input_model.groupdq = np.broadcast_to(R_input_model.groupdq, (1,) +
-                                                  R_input_model.groupdq.shape)
-
-            input_model.err = np.broadcast_to(R_input_model.err, (1,) +
-                                              R_input_model.err.shape)
-
-            input_model.refout = np.broadcast_to(R_input_model.refout, (1,) +
-                                                 R_input_model.refout.shape)
-
+        with rdd.open(input, mode='rw') as input_model:
+            input_model.data = input_model.data[np.newaxis, :]
+            input_model.data.dtype=np.float32
+            input_model.groupdq = input_model.groupdq[np.newaxis, :]
             max_cores = self.maximum_cores
-            readnoise_filename = self.get_reference_file(input_model,
-                                                         'readnoise')
+            readnoise_filename = self.get_reference_file(input_model, 'readnoise')
             gain_filename = self.get_reference_file(input_model, 'gain')
 
             log.info('Using READNOISE reference file: %s', readnoise_filename)
-            readnoise_model = datamodels.ReadnoiseModel(readnoise_filename)
+            readnoise_model = rdd.open(readnoise_filename, mode='rw')
             log.info('Using GAIN reference file: %s', gain_filename)
-            gain_model = datamodels.GainModel(gain_filename)
-
-            # Try to retrieve the gain factor from the gain reference file.
-            # If found, store it in the science model meta data, so that it's
-            # available later in the gain_scale step, which avoids having to
-            # load the gain ref file again in that step.
-            if gain_model.meta.exposure.gain_factor is not None:
-                input_model.meta.exposure.gain_factor = \
-                    gain_model.meta.exposure.gain_factor
```

Comment:
I assume this is no longer needed due to DataModel refactoring ?

Reply:
I was wondering - there is no `gain_factor` in the gain schema. Did we miss it or is it somewhere else? @ddavis-stsci ?

Reply:
**Adding our conversation from slack for reference:**
dencheva:
gain models in jwst have meta.exposure.gain_factor attribute. This is missing from the roman one. Is this supposed to be in the data array for Roman? The documentation for gain reference file does not mention gain_factor.

phuwe  2:30 PM
gain_factor is in exposure in RAD

ddavis:house_with_garden:  2:35 PM
I don’t think we have gain_factors in Roman. If I recall correctly this is to account for non-standard gain factors when used in certain JWST modes. I suppose Roman can use them but so far we have no use case for a gain_factor

perry  2:36 PM
If it is needed by the existing JWST code, we can just generate a value of 1 in the converter.

ddavis:house_with_garden:  2:37 PM
The value of 1 should be fine if we need it. (edited) 

dencheva:house_with_garden:  2:48 PM
There's no need for that, it's used within the pipeline code (i.e. it's mission specific)  and in this case can be ignored. I just wanted to confirm we did not miss anything.
2:50
In jwst it's used in gain_scale. AFAIK we are not going to run gain_scale for roman but it's still listed in the docs as one of the steps
2:51
Or perhaps I misunderstand its use. @grumm may not better

grumm:slightly_smiling_face:  3:29 PM
Back in February  we thought this was needed (GH PR 120: Create gain factor keyword schema; Gain factor keyword needed. RCAL#91) . I am not aware of any more recent changes regarding whether it is needed.